controllers/azuremachine_reconciler.go

```diff
@@ -60,17 +54,19 @@ func newAzureMachineService(machineScope *scope.MachineScope) (*azureMachineServ
 	}
 
 	return &azureMachineService{
-		scope:                machineScope,
-		inboundNatRulesSvc:   inboundnatrules.New(machineScope),
-		networkInterfacesSvc: networkinterfaces.New(machineScope, cache),
-		virtualMachinesSvc:   virtualmachines.New(machineScope),
-		roleAssignmentsSvc:   roleassignments.New(machineScope),
-		disksSvc:             disks.New(machineScope),
-		publicIPsSvc:         publicips.New(machineScope),
-		tagsSvc:              tags.New(machineScope),
-		vmExtensionsSvc:      vmextensions.New(machineScope),
-		availabilitySetsSvc:  availabilitysets.New(machineScope, cache),
-		skuCache:             cache,
+		scope: machineScope,
+		services: []azure.Reconciler{
+			publicips.New(machineScope),
+			inboundnatrules.New(machineScope),
+			networkinterfaces.New(machineScope, cache),
+			availabilitysets.New(machineScope, cache),
+			disks.New(machineScope),
```

Comment:
this is the same order as Reconcile() below except for disks that were not present before in Reconcile (it's a no-op) but needs to be deleted _after_ the VM (hence needs to be before VMs in the list)

Reply:
What was our behavior before this change? Would the disk get deleted anyway?

Reply:
There was a disk delete call https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2146/files#diff-5a6d880b9a0be00329894fc2f52f15827dd6bdfe6b60aaec660728661023f41fL142 but no disk reconcile call. Disk reconcile is a no-op so adding it does not change behavior.